src/logic.py

```diff
@@ -176,6 +177,25 @@ def assign_author(self) -> tuple[bool, str]:
             True,
             f'Author {self.author} successfully assigned to PR, "{self.config.awaiting_update_label}" label added',
         )
+    
+    def parse_magic_string(self) -> str:
+        """
+            Parses the commit body to find magic string
+        """
+
+        _username = None
+        search_username = re.findall(r'primary-reviewer:\s@[a-zA-Z][a-zA-Z0-9-_\.]{1,20}$', self.comment.body)
+
+
+        if(len(search_username)):
+            # found magic comment
+            _username = search_username[0].split('@')[1]
+        else:
+            _username = random.choice(self.reviewers)
+            self.comment.body += f"\n\n{_username}"
+        
+        return _username
```

Comment:
you need to check the reviewer is in `self.reviewers` and return an error if not.